connection.go

```diff
@@ -554,9 +554,23 @@ func (sc *snowflakeConn) FetchResult(ctx context.Context, qid string) (driver.Ro
 // go sql library but is exported to clients who can make use of this
 // capability explicitly.
 func (sc *snowflakeConn) WaitForQueryCompletion(ctx context.Context, qid string) error {
+	// if the query is already done, we dont need to wait for it 
+	complete, err := sc.checkIfComplete(ctx, qid) 
+	if complete {
+		return err
+	}
 	return sc.blockOnQueryCompletion(ctx, qid)
 }
```

Comment:
I dont really know why this would be needed but I snowflake does it [here](https://github.com/sigmacomputing/gosnowflake/blob/master/connection.go#:~:text=//%20check%20the%20query,%7D), and I don't see a reason not to do it. It doesnt add any latency